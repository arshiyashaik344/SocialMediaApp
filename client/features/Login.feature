Feature: Login Page User Details

Scenario: The one where user tries LogIn with valid credentials
 Given user navigates to the login form page
    When user enters "arshiyas" in the userName field
    And user enters "arshi" in the Password field
    And user clicks on Login button
    Then posts page with userName "Arshiyas" appears

 