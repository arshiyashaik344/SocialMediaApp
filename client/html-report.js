var reporter = require('cucumber-html-reporter');
var options = {
  theme: 'bootstrap',
  jsonFile: 'reports/report.json',
  output: 'reports/cucumber_report.html',
  reportSuiteAsScenarios: true,
  scenarioTimestamp: true,
  launchReport: true,
  metadata: {
    Title: 'Sample Report',
    Browser: 'chrome',
    Platform: 'Linux'
  }
};
reporter.generate(options);
