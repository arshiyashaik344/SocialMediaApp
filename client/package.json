{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@apollo/client": "^3.4.17",
    "@apollo/react-hooks": "^4.0.0",
    "apollo-link-context": "^1.0.18",
    "apollo-link-ws": "^1.0.20",
    "cucumber-html-reporter": "^5.5.0",
    "gherkin-testcafe": "^5.2.0",
    "graphql": "^15.7.2",
    "graphql-tag": "^2.10.1",
    "jwt-decode": "^2.2.0",
    "moment": "^2.29.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-paginate": "^8.1.0",
    "react-router-dom": "^5.3.0",
    "react-scripts": "^3.0.1",
    "semantic-ui-css": "^2.4.1",
    "semantic-ui-react": "^2.0.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "dev": "webpack serve",
    "build": "webpack",
    "test": "./node_modules/@cucumber/cucumber/bin/cucumber-js --format json:./reports/report.json",
    "eject": "react-scripts eject",
    "lint": "eslint .",
    "lint:fix": "eslint --fix",
    "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.16.7",
    "@babel/plugin-syntax-jsx": "^7.16.7",
    "@babel/preset-env": "^7.16.7",
    "@babel/preset-react": "^7.16.7",
    "@cucumber/cucumber": "^7.3.2",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^13.5.0",
    "babel-jest": "^27.4.6",
    "babel-loader": "^8.2.3",
    "css-loader": "^6.5.1",
    "eslint": "^7.32.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react": "^7.28.0",
    "html-webpack-plugin": "^4.5.2",
    "jest": "^27.5.1",
    "jest-environment-jsdom": "^27.4.6",
    "prettier": "^2.5.1",
    "react-test-renderer": "^17.0.2",
    "resolve-url-loader": "^5.0.0",
    "sass-loader": "^12.4.0",
    "style-loader": "^3.3.1",
    "url-loader": "^4.1.1",
    "vue-jest": "^3.0.7",
    "webpack": "^5.65.0",
    "webpack-cli": "^4.9.1",
    "webpack-dev-server": "^4.7.4"
  },
  "jest": {
    "verbose": true,
    "testEnvironment": "jsdom"
  }
}
