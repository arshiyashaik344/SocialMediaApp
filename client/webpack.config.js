const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');

module.exports = {
  entry: path.join(__dirname, 'src', 'index.js'),
  output: {
    path: path.resolve(__dirname, 'dist')
  },
  mode: 'development',
  module: {
    rules: [
      {
        test: /\.?js$/,
        exclude: /node_modules/,
        use: {
          loader: 'babel-loader',
          options: {
            presets: [
              '@babel/preset-env',
              '@babel/preset-react',
              {
                plugins: ['@babel/plugin-syntax-jsx']
              }
            ]
          }
        }
      },
      {
        test: /\.css$/i,
        use: [
          {
            loader: 'style-loader'
          },
          {
            loader: 'css-loader',
            options: {
              modules: true,
              sourceMap: true
            }
          },
          {
            loader: 'resolve-url-loader',
            options: {
              removeCR: true
            }
          }
        ],
        include: [path.join(__dirname, 'src'), /node_modules/]
      },
      // {
      //   test: /\.s[a|c]ss$/,
      //   use: [
      //     {
      //       loader: 'sass-loader'
      //     },
      //     {
      //       loader: 'style-loader'
      //     },
      //     {
      //       loader: 'css-loader'
      //     }
      //   ],
      //   include: [path.join(__dirname, 'src'), /node_modules/]
      // },
      {
        test: /\.(png|jpg|woff|svg|eot|ttf|woff2)$/,
        use: {
          loader: 'url-loader',
          options: {
            limit: 1000,
            name: 'assets/img/[name].[ext]'
          }
        }
      }
    ]
  },
  plugins: [
    new HtmlWebpackPlugin({
      template: path.join(__dirname, 'public', 'index.html')
    })
  ]
};
