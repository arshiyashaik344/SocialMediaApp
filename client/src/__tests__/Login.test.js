/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
// import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/jest-dom';
import { MockedProvider } from '@apollo/client/testing';
import TestRenderer from 'react-test-renderer';
import React from 'react';

import Login from '../pages/Login';
import { render } from 'react-dom';

const mocks = [];

describe('Login Component', () => {
  it('rendered input', () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Login />
      </MockedProvider>
    );

    const tree = component.toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
    const instance = component.root;
    expect(instance.findByProps({ htmlFor: 'user-name' }).children).toEqual([
      'Username'
    ]);
    // instance.findByProps({ placeholder: 'Username..' }, 'arshiyas');
    // instance.findByProps({ placeholder: 'Password..' }, 'arshi');
    // const username = instance.findByProps({ className: 'ui form' }).children[1];
    // username.findByProps({ value: '' }, 'arshiyas');
    // console.log(username);

    // expect(instance.findByProps({ className: 'ui primary button' }).click);
    // console.log(instance.type);
  });

  // test('DOM testing', () => {
  //   const login = render(
  //     <MockedProvider>
  //       <Login />
  //     </MockedProvider>
  //   );
  //   screen.debug();
  // });
});

// test('Login Component', () => {
//   render(<Login />);
//   const buttonElement = screen.getByTestId('input');
//   expect(buttonElement).toBeInTheDocument();
//   // expect(buttonElement).toHaveTextContent('Clear');
// });

// ******* Generic Testing **********

// render(<Login />);
// const { getByTestId } = render(<Login />);
// const heading = screen.getByText('Login');
// expect(heading).toBeInTheDocument();
// const input = getByTestId('input');
// expect(input).toBeTruthy();
// screen.debug();

// userEvent.type(screen.getByPlaceHolderText(/Username../i), 'arshiyas');
// userEvent.type(screen.getByPlaceHolderText(/Password../i), 'arshi');

// expect(screen.findByRole('button', { type: 'submit' })).toBeEnabled();
