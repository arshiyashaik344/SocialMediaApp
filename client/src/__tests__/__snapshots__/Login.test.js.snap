// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component rendered input 1`] = `
<div
  className="form-container"
>
  <form
    className="ui form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <h1>
      Login
    </h1>
    <div
      className="field"
    >
      <label
        htmlFor="user-name"
      >
        Username
      </label>
      <div
        className="ui input"
        data-testid="input"
      >
        <input
          aria-describedby={null}
          id="user-name"
          name="username"
          onChange={[Function]}
          placeholder="Username.."
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="user-password"
      >
        Password
      </label>
      <div
        className="ui input"
        data-testid="password"
      >
        <input
          aria-describedby={null}
          id="user-password"
          name="password"
          onChange={[Function]}
          placeholder="Password.."
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      className="ui primary button"
      id="login-button"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
